sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -570,18 +570,44 @@ class Analyzer(
           s"Grouping sets size cannot be greater than ${GroupingID.dataType.defaultSize * 8}")
       }
 
+      // For struct field, it will be resolve as Alias(GetStructField, name),
+      // In Aggregate/GroupingSets this behavior will cause the same struct fields
```

Comment:
Can we fix the problem earlier in `ResolveReferences`? e.g.·
```
case a: Aggregate =>
  val resolvedGroupingExprs = CleanupAliases.trimAliases(
    a.groupingExpressions.map(resolveExpressionTopDown(_, a))
  val resolvedAggExprs = a.aggregateExpressions.map(resolveExpressionTopDown(_, a)
  a.makeCopy(resolvedGroupingExprs, resolvedAggExprs, a.child)
```

Reply:
Similar like  https://github.com/apache/spark/pull/28490#discussion_r470555710.
Need to try if it can solve all problem use `CleanupAlias.trimAliases()`.

Reply:
It's simpler to fix the problem earlier. Let's try it.

Reply:
> It's simpler to fix the problem earlier. Let's try it.

Ok, update later.

Reply:
cc @cloud-fan 
Updated. since it's special, we can't just use CleanupAlias.trimAlias()
Add a new method to fit for GetStructField